sklearn/preprocessing/_encoders.py

```diff
@@ -418,15 +472,64 @@ def fit(self, X, y=None):
 
         self._handle_deprecations(X)
 
+        # If we have both dropped columns and ignored unknown
+        # values, there will be ambiguous cells. This creates difficulties
+        # in interpreting the model.
+        if self.drop is not None and self.handle_unknown != 'error':
+            raise ValueError(
+                "`handle_unknown` cannot be 'ignore' when the drop parameter "
+                "is specified, as this will create ambiguous cells")
```

Comment:
given that you needed to add it in two places, maybe it is worth making a helper method of it like `self._validate_keywords()` (that can also include the validation of `self.handle_unknown` that is now a few lines above, as that is also repeated)